### PR TITLE
Azure Maps Java Timezone SDK

### PR DESCRIPTION
**Pull request by khmic5**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/maps/azure-maps-timezone/src/samples/java/com/azure/maps/timezone/samples/ConvertWindowsTimezoneToIana.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        // AzureKeyCredential keyCredential = new AzureKeyCredential(System.getenv("SUBSCRIPTION_KEY"));
+        // builder.credential(keyCredential);
+
+        // Authenticates using Azure AD building a default credential
+        // This will look for AZURE_CLIENT_ID, AZURE_TENANT_ID, and AZURE_CLIENT_SECRET env variables
+        DefaultAzureCredential tokenCredential = new DefaultAzureCredentialBuilder().build();
+
+        builder.credential(tokenCredential);
+        builder.timezoneClientId(System.getenv("MAPS_CLIENT_ID"));
+        builder.httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS));
```
Comment (lmolkova) [inline sdk/maps/azure-maps-timezone/src/samples/java/com/azure/maps/timezone/samples/ConvertWindowsTimezoneToIana.java:28]: please avoid using BODY_AND_HEADERS in this and other samples, you can remove `builder.httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS));` altogether or just have a specific example for troubleshooting section.

File: sdk/maps/azure-maps-timezone/src/samples/java/com/azure/maps/timezone/samples/ConvertWindowsTimezoneToIana.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        TimezoneClient client = builder.buildClient();
```
Comment (lmolkova) [inline sdk/maps/azure-maps-timezone/src/samples/java/com/azure/maps/timezone/samples/ConvertWindowsTimezoneToIana.java:29]: nit:, please use fluent style (`new Builder().foo().bar().build()`) - we use it for other SDKs

File: sdk/maps/azure-maps-timezone/src/main/java/com/azure/maps/timezone/models/TimezoneIdOptions.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        return this;
+    }
+
+    /**
+     * get accept language
+     * @return the accept language. Specifies the language code in which the timezone names should be returned. If no language
+     *     code is provided, the response will be in "EN". Please refer to [Supported
+     *     Languages](https://docs.microsoft.com/azure/azure-maps/supported-languages) for details.
+     */
+    public String getAcceptLanguage() {
```
Comment (lmolkova) [inline sdk/maps/azure-maps-timezone/src/main/java/com/azure/maps/timezone/models/TimezoneIdOptions.java:51]: we have `TimezoneCoordinateOptions.getLanguage()`, please use consistent API names

```suggestion
    public String getLanguage() {
```

File: sdk/maps/azure-maps-timezone/src/main/java/com/azure/maps/timezone/models/TimezoneIdOptions.java [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    }
+
+    /**
+     * set accept language
+     * @param acceptLanguage Specifies the language code in which the timezone names should be returned. If no language
+     *     code is provided, the response will be in "EN". Please refer to [Supported
+     *     Languages](https://docs.microsoft.com/azure/azure-maps/supported-languages) for details.
+     * @return TimezoneIdOptions
+     */
+    public TimezoneIdOptions setAcceptLanguage(String acceptLanguage) {
```
Comment (lmolkova) [inline sdk/maps/azure-maps-timezone/src/main/java/com/azure/maps/timezone/models/TimezoneIdOptions.java:62]:
```suggestion
    public TimezoneIdOptions setLanguage(String acceptLanguage) {
```

**Review by lmolkova — COMMENTED** [review]
Left a few minor comments, looks good overall!

File: sdk/maps/azure-maps-timezone/pom.xml [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+      <version>1.32.0</version> <!-- {x-version-update;com.azure:azure-core;dependency} -->
+    </dependency>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-core-http-netty</artifactId>
+      <version>1.12.5</version> <!-- {x-version-update;com.azure:azure-core-http-netty;dependency} -->
+    </dependency>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-core-http-okhttp</artifactId>
```
[the remark this answers is not in the document]
Comment (khmic5) [reply sdk/maps/azure-maps-timezone/pom.xml:68]: If this is removed the live tests don't pass due to "PreconditionViolation Configuration error: You must configure at least one set of arguments for this @ParameterizedTest"

**Review by lmolkova — APPROVED** [review]

**Pull request merged**